dashboard/test/integration/pardot_test.rb

```diff
@@ -0,0 +1,87 @@
+require 'test_helper'
+require 'cdo/contact_rollups'
+
+PEGASUS_TEST_DB_NAME = "pegasus_#{Rails.env}"
+COLUMN_NAME_TO_INDEX_MAP = { "roles": 0, "ages_taught": 1 }.freeze
+
+class PardotTest < ActiveSupport::TestCase
+  def test_empty_contacts
+    # Test the rollup process with an empty database
+    rollups_test_helper 0, []
+  end
+
+  def test_teachers
+    # Create teacher 1 with no section
+    @teacher1 = create(:teacher, email: "rolluptestteacher1@code.org")
+
+    # Create teacher 2 with one section and one student
+    @teacher2 = create(:teacher, email: "rolluptestteacher2@code.org")
+    create_sections_helper @teacher2, [[{age: 6}]]
+
+    # Create teacher 3 with one section and multiple students
+    @teacher3 = create(:teacher, email: "rolluptestteacher3@code.org")
+    @teacher3_section = create(:section, user: @teacher3)
+    create_sections_helper @teacher3, [[{age: 6}, {age: 10}, {age: 14}, {age: 10}]]
+
+    # Create teacher 4 with three sections and multiple students
+    @teacher4 = create(:teacher, email: "rolluptestteacher4@code.org")
+    create_sections_helper @teacher4,
+      [
+        [{age: 14}, {age: 10}, {age: 11}, {age: 10}, {age: 10}, {age: 11}],
+        [{age: 9}, {age: 11}, {age: 9}],
+        [{age: 15}]
+      ]
+
+    # Expected values for ages taught is the union of all unique values of student ages,
+    # no duplicates, in ascending sorted order
+    expected_values = [
+      { "rolluptestteacher1@code.org": { "roles": "Teacher", "ages_taught": nil }},
+      { "rolluptestteacher2@code.org": { "roles": "Teacher", "ages_taught": "6" }},
+      { "rolluptestteacher3@code.org": { "roles": "Teacher", "ages_taught": "6,10,14" }},
+      { "rolluptestteacher4@code.org": { "roles": "Teacher", "ages_taught": "9,10,11,14,15" }}
+    ]
+
+    rollups_test_helper 4, expected_values
+  end
+
+  private
+
+  def rollups_test_helper(expected_count, expected_values)
+    # Run the rollup process
+    ContactRollups.build_contact_rollups
+
+    # Should now have expected_count records in daily rollups table, and still none in main rollups table
+    assert ActiveRecord::Base.connection.execute("select count(*) from pegasus_test.contact_rollups_daily").first[0] == expected_count
+    assert ActiveRecord::Base.connection.execute("select count(*) from pegasus_test.contact_rollups").first[0] == 0
+
+    # Verify expected values in contacts_rollup_daily
+    expected_values.each do |expected_values_hash|
+      email = expected_values_hash.keys.first
+      email_sanitized = ActiveRecord::Base.sanitize(expected_values_hash.keys.first)
```

Comment:
I suppose it can't hurt, but I don't think you need to sanitize values that you provide in the test... unless you're worried about your own SQL injection attack ;)